lib/modules.nix

```diff
@@ -730,9 +730,16 @@ let
             else res.options;
         in opt.options // res //
           { declarations = res.declarations ++ [opt._file];
+            # In the case of modules that are generated dynamically, we won't
+            # have exact declaration lines; fall back to just the file being
+            # evaluated.
+            declarationsWithLocations = res.declarationsWithLocations
```

Comment:
"Location" has also been defined to be the option path, option path prefix or that kind of thing.

Also these are only the locations, so I think `declarationPositions` would be a better name.

Reply:
This is mirroring definitionsWithLocations. I'm open to changing it, but it would make these inconsistent with each other. Although to be fair that one is only file granularity... like definitions. 

This API could really go either way.

Reply:
> it would make these inconsistent with each other.

But they are actually already inconsistent. The new one doesn't also have whatever is at the location, as in the value of the key-value maps that these represent.

Now this raises the question: _should_ we have an option attribute that has both the declarations and its location in the same attrset, like `definitionsWithLocations`?

I think we should not. What you wrote is a simple solution with a minimal impact on performance.
Transposing the two lists into a single list of "key value" pairs would add more overhead for an infrequently used feature.

> it seems to be in the noise

Individual incremental changes never raise concern if we have sufficient noise. We do.
I prefer to pick a more reliable metric, such as the memory related stats in the evaluation report commit status (such as [this one, current as of writing](https://github.com/NixOS/nixpkgs/pull/249243/checks?check_run_id=16050873293))
It isn't without flaws either. Total memory tends to be rounded to at least 16 MB, it's nondeterministic. The exact versions compared are unclear (but probably ok if the branch is recent). Total allocations is probably the best metric, as it correlates well with "work" and therefore time.

Nonetheless, the deterioration is small and I believe it is offset by the improved usability, but not by a large margin. For this reason I think we should stick to the current simple implementation and apply the rename to differentiate it from `definitionsWithLocations`.

Reply:
alright, i have applied the rename.